third_party/tsl/third_party/spirv_llvm_translator/spirv_llvm_translator.patch

```diff
@@ -0,0 +1,20 @@
+diff --git a/lib/SPIRV/SPIRVInternal.h b/lib/SPIRV/SPIRVInternal.h
+index a828add8..924e13b4 100644
+--- a/lib/SPIRV/SPIRVInternal.h
++++ b/lib/SPIRV/SPIRVInternal.h
+@@ -179,11 +179,12 @@ typedef SPIRVMap<Op, Op, IntBoolOpMapId> IntBoolOpMap;
+   "-v512:512:512-v1024:1024:1024"
+
+ enum SPIRAddressSpace {
+-  SPIRAS_Private,
++  SPIRAS_Generic,
+   SPIRAS_Global,
+-  SPIRAS_Constant,
++  SPIRAS_Internal,
+   SPIRAS_Local,
+-  SPIRAS_Generic,
++  SPIRAS_Constant,
++  SPIRAS_Private,
```

Comment:
Spir backend uses different addrspace representations [link](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/blob/0aab1249acbfb8d206c24988f0974638c290e931/lib/SPIRV/SPIRVInternal.h#L181) with nvptx backend [link](https://llvm.org/docs/NVPTXUsage.html#address-spaces).  We reorder the enum value here so that we can make xla llvm codegen simple(avoiding changing addrspace based on device backend everywhere)

Reply:
This is useful context for people who are unfamiliar with the backend. 

Could you please add this explanation in this patch file (between line 2 and 3)? I believe the patch tool will ignore anything above the `--- a/lib/SPIRV/SPIRVInternal.h` line. E.g.,
```
diff --git a/lib/SPIRV/SPIRVInternal.h b/lib/SPIRV/SPIRVInternal.h
index a828add8..924e13b4 100644

<Explanation here>

--- a/lib/SPIRV/SPIRVInternal.h
```